### ISSUE TITLE
add (automatic) pkgdown pages

### ISSUE THREAD

**Issue by goldingn** [zoonproject/zoon#346]
For some reason, I only just spotted [pkgdown](https://github.com/hadley/pkgdown) which automatically produces [pretty web pages](http://hadley.github.io/pkgdown/) for R package docs, using github pages. It would be great to implement this for both the zoon package and the modules repo, so we have nice pages to send people to.

Generating the pages is fairly easy, so long as we remember to run `pkgdown::build_site()` on each commit. However we're liable to forget, and it would be awesome if the docs could be updated automatically on each successful master Travis build. That should be doable, via [this method](https://gist.github.com/domenic/ec8b0fc8ab45f39403dd).

Given Travis will be pushing, we should probably stick to using the `gh-pages` branch for github pages, rather than the `docs` directory on `master` (as suggested in `pkgdown`).

I'll have a punt at this now (on a new branch) so this is just to document what I'm up to and track progress. There will probably be some tidying needed before the pages are pretty-looking.

**Comment by goldingn** [comment]
example zoon site now up at: http://goldingn.github.io/zoon

**Comment by goldingn** [comment]
The automated build setup looks like it should work, but my travis builds keep failing due to issues in occurrence modules; presumably the GBIF API being flaky as usual.

Will test it some more when they start passing.

**Comment by goldingn** [comment]
and here's the modules page: http://goldingn.github.io/modules/
No automation set up on that one, I'll put that in when I get the other one working.

**Comment by AugustT** [comment]
This all looks great!

**Comment by AugustT** [comment]
So this works because GitHub knows that a branch called `gh-pages` is a website? This is all very clever!

**Comment by AugustT** [comment]
I'm removing SpOcc from tests in the future

**Comment by goldingn** [comment]
Yup, you can point pages either to the gh-pages branch, to a folder 'docs'  on master, or to master itself. The auto-building is the tricky bit.

**Comment by AugustT** [comment]
yeah... I saw the the .sh script.... looks taxing!

**Comment by goldingn** [comment]
note to future self: make sure the ssh keypair doesn't have a passphrase :/
getting there ...

**Comment by goldingn** [comment]
More notes for future me:

1. create a `gh-pages` branch
1. locally create an ssh keypair named *deploy_key* (private: `deploy_key`, public: `deploy_key.pub`) with no passphrase (follow [these](https://help.github.com/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent/) instructions, but just hit return for the passphrase)
1. from the repo's github page, navigate to *settings/deploy keys* and paste the contents of `deploy_key.pub` into a new deploy key, wand check the *read/write access* box
1. encrypt the private key with the travis CLI (`deploy_key` -> `deploy.enc`) as described [here](https://gist.github.com/domenic/ec8b0fc8ab45f39403dd#get-encrypted-credentials)
1. commit the *Travis-encrypted private* key (`deploy_key.enc`) to `master`
1. copy the `.deploy_pkgdown.sh` script as-is to `master`.
1. modify `.travis.yml`'s `after_success:` on `master` to add ` - bash .deploy_pkgdown.sh` 
1. modify `.travis.yml` on `master` to install `pkgdown` and `rmarkdown` from github, and `libmagick++-dev` from apt
1. pray you haven't forgotten something

**Comment by goldingn** [comment]
OK, this now auto pushes: https://github.com/goldingn/zoon/commit/3a5498ff347a17188a84d6a34783301b0ef8a377

**Comment by goldingn** [comment]
However images in the vignettes aren't loading since the HTML is being populated absolute file paths (previously my local files, now relative to the Travis build environment: https://github.com/goldingn/zoon/commit/3a5498ff347a17188a84d6a34783301b0ef8a377#diff-304b279dbd02f2a2d35ad7db3cc07ae1R112)

Need to look into how to make that portable

**Comment by goldingn** [comment]
also, we can use any one of these themes: https://gallery.shinyapps.io/117-shinythemes/ any favourites @AugustT @GregMci?

**Comment by AugustT** [comment]
I like the look of Sandstone but defer to the expertise of @GregMci !

**Comment by goldingn** [comment]
yeah, sandstone is nice and light and fits in with the zoon colour scheme fairly well

**Comment by goldingn** [comment]
The vignette images issue should be fixed now: https://github.com/hadley/pkgdown/issues/172

**Comment by goldingn** [comment]
OK, next I'll:

 - [ ] use sandstone scheme
 - [ ] PR into zoon/master

I'll leave the modules page for now, until we know what's happening with the website
